### PR TITLE
Update dependencies to latest stable versions

### PR DESCRIPTION
**Pull request by Kami**
This pull request updates various dependencies to latest stable versions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -383,7 +383,8 @@ requirements: virtualenv .sdist-requirements install-runners
 	$(VIRTUALENV_DIR)/bin/pip install "six==1.11.0"
 
 	# Fix for Travis CI caching issue
-	$(VIRTUALENV_DIR)/bin/pip uninstall -y "pytz"
+	$(VIRTUALENV_DIR)/bin/pip uninstall -y "pytz" || echo "not installed"
+	$(VIRTUALENV_DIR)/bin/pip uninstall -y "python-dateutil" || echo "not installed"
```
Comment (Kami) [inline Makefile]: Need more workarounds for what looks like Travis caching and pip weirdness.

Comment (armab) [reply Makefile]: BTW related to that, I got this on local dev env unable to build from sources on a clean Ubuntu Xenial VM:

```
Generating requirements.txt for contrib/runners/action_chain_runner
===========================================================
Requirements written to: contrib/runners/action_chain_runner/requirements.txt
# Fix for Travis CI race
virtualenv/bin/pip install "six==1.11.0"
Requirement already satisfied: six==1.11.0 in ./virtualenv/lib/python2.7/site-packages
You are using pip version 9.0.3, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
# Fix for Travis CI caching issue
virtualenv/bin/pip uninstall -y "pytz"
Uninstalling pytz-2018.7:
  Successfully uninstalled pytz-2018.7
Traceback (most recent call last):
  File "virtualenv/bin/pip", line 11, in <module>
    sys.exit(main())
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/__init__.py", line 248, in main
    return command.main(cmd_args)
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/basecommand.py", line 252, in main
    pip_version_check(session)
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/utils/outdated.py", line 102, in pip_version_check
    installed_version = get_installed_version("pip")
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 838, in get_installed_version
    working_set = pkg_resources.WorkingSet()
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 644, in __init__
    self.add_entry(entry)
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 700, in add_entry
    for dist in find_distributions(entry, True):
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1949, in find_eggs_in_zip
    if metadata.has_metadata('PKG-INFO'):
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1463, in has_metadata
    return self.egg_info and self._has(self._fn(self.egg_info, name))
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1823, in _has
    return zip_path in self.zipinfo or zip_path in self._index()
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1703, in zipinfo
    return self._zip_manifests.load(self.loader.archive)
  File "/st2/virtualenv/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1643, in load
    mtime = os.stat(path).st_mtime
OSError: [Errno 2] No such file or directory: '/st2/virtualenv/local/lib/python2.7/site-packages/pytz-2018.7-py2.7.egg'
Makefile:363: recipe for target 'requirements' failed
make: *** [requirements] Error 1

```

Comment (Kami) [reply Makefile]: I had similar issues here, even if I used old version of pytz. Looks like some pip related weirdness.

I got it to work by manually uninstalling python-dateutil and pytz once make requirements failed and then ran it again.

Comment (Kami) [reply Makefile]: Just tested it again with a fresh vm and a new virtualenv and it seems to work fine.

**Pull request merged**